contrib/lkt/language/parser.py

```diff
@@ -2078,9 +2082,14 @@ def called_decl():
 
         # Implement special resolution for calls to objects via __call__
         called_decl = Var(
-            refd_decl.cast(T.BaseValDecl)
-            ._.get_type._.get_fun('__call__')
-            .then(lambda a: a, default_val=refd_decl)
+            refd_decl.match(
+                # Don't try to look for __call__ on function decls
+                lambda _=FunDecl: refd_decl,
+                lambda v=BaseValDecl:
+                v._.get_type._.get_fun('__call__')
+                .then(lambda a: a, default_val=refd_decl),
+                lambda _: refd_decl,
+            )
```

Comment:
Can you add a testcase for the cases where we had crashes and don’t anymore?

Reply:
Erf I had one when I did this... Don't know where I put it... I'll manage!